spec/datadog/kit/events_spec.rb

```diff
@@ -10,72 +10,95 @@
 RSpec.describe Datadog::Kit::Events do
   subject(:trace_op) { Datadog::Tracing::TraceOperation.new }
 
-  describe '#track_login' do
-    context 'login success' do
-      it 'sets event tracking key on trace' do
-        trace_op.measure('root') do
-          described_class.track_login(trace_op, user: { id: '42' }, success: true)
-        end
-        trace = trace_op.flush!
-        expect(trace.send(:meta)).to include('appsec.events.users.login.success.track' => 'true')
+  describe '#track_login_success' do
+    it 'sets event tracking key on trace' do
+      trace_op.measure('root') do
+        described_class.track_login_success(trace_op, user: { id: '42' })
       end
+      trace = trace_op.flush!
+      expect(trace.send(:meta)).to include('appsec.events.users.login.success.track' => 'true')
+    end
 
-      it 'sets user id on trace' do
-        trace_op.measure('root') do
-          described_class.track_login(trace_op, user: { id: '42' }, success: true)
-        end
-        trace = trace_op.flush!
-        expect(trace.send(:meta)).to include('usr.id' => '42')
+    it 'sets user id on trace' do
+      trace_op.measure('root') do
+        described_class.track_login_success(trace_op, user: { id: '42' })
       end
+      trace = trace_op.flush!
+      expect(trace.send(:meta)).to include('usr.id' => '42')
+    end
 
-      it 'sets other keys on trace' do
-        trace_op.measure('root') do
-          described_class.track_login(trace_op, user: { id: '42' }, success: true, foo: 'bar')
-        end
-        trace = trace_op.flush!
-        expect(trace.send(:meta)).to include('usr.id' => '42', 'appsec.events.users.login.success.foo' => 'bar')
+    it 'sets other keys on trace' do
+      trace_op.measure('root') do
+        described_class.track_login_success(trace_op, user: { id: '42' }, foo: 'bar')
       end
+      trace = trace_op.flush!
+      expect(trace.send(:meta)).to include('usr.id' => '42', 'appsec.events.users.login.success.foo' => 'bar')
```

Comment:
Minor: This bit seems to repeat quite a lot

```ruby
trace = trace_op.flush!
trace.send(:meta)
```

consider maybe making it a let?

Reply:
Hmm, I have to think about it, I'm not to keen on `let` that do "actions", as it creates magic that gets unwieldy.

I've suffered a lot through that in some specs (IIRC Rack or Sinatra ones). If I can figure something sane quickly I'll do it, otherwise I'd favour explicitness in tests because it can definitely enhance readability even though it's repetitive.

Reply:
OK I've done that.